h2/utilities.py

```diff
@@ -215,6 +225,24 @@ def _reject_uppercase_header_fields(headers, hdr_validation_flags):
         yield header
 
 
+def _reject_surrounding_whitespace(headers, hdr_validation_flags):
+    """
+    Raises a ProtocolError if any header name or value is surrounded by
+    whitespace characters.
+    """
+    for header in headers:
+        if header[0] and ((header[0][0] in _WHITESPACE) or
+                          (header[0][-1] in _WHITESPACE)):
+            raise ProtocolError(
+                "Received header name surrounded by whitespace %r" % header[0])
+        if header[1] and ((header[1][0] in _WHITESPACE) or
```

Comment:
@Lukasa If I don't check for empty headers the test at https://github.com/python-hyper/hyper-h2/blob/7a80186834fd344628941bc39e333d0ea108652e/test/test_invalid_headers.py#L211 fails. Should I raise a ProtocolError here if the headers are empty?